elegant/gui/general_pose_annotator.py

```diff
@@ -0,0 +1,95 @@
+import freeimage
+import pathlib
+import pickle
+import pkg_resources
+from PyQt5 import Qt
+import numpy
+from functools import partial
+
+from ris_widget import ris_widget
+from ris_widget import shared_resources
+
+from elegant.gui import pose_annotation
+from elegant import worm_widths
+from elegant import worm_spline
+
+from zplib.image import colorize
+
+def save_annotations(rw):
+    """Save the pose annotations as pickle files into the parent directory.
+    A pickle file is created for each page in the flipbook with the name of the first image in the 
+    flipbook_page list as the base for the pickle file name.
+    """
+    for fp in rw.flipbook_pages:
+        annotations = fp.annotations
+        path = pathlib.Path(fp[0].name)
+        save_path = path.parent/(path.name.split(".")[0]+".pickle") 
+        pickle.dump(annotations, open(save_path, 'wb'))
+
+def load_metadata(rw):
+    """Loads annotation metadata from a pickle file for each flipbook page.
+    NOTE: the metadata is only loaded if there is no current pose annotations.
+    """
+    for fp in rw.flipbook_pages:
+        center_tck, width_tck = fp.annotations['pose']
+        if center_tck is None and width_tck is None:
+            path = pathlib.Path(fp[0].name)
+            img_name = path.name.split('.')[0]
+            glob = path.parent.glob(img_name+('.pickle'))
+            
+            for ann_path in glob:
+                annotations = pickle.load(open(ann_path, 'rb'))
+                fp.annotations = annotations
+
+    rw.annotator.update_fields()
+
+def save_warped_images(rw):
+    """Save the warped images into the same parent directory as the images in the flipbook.
+    If there is a width tck, the saved images have a transparent channel with the worm mask.
+    The warped images will be saved with the same base name as the first image in the flipbook_page list
+    with "-straightening.png" afterwards.
+
+    Example:
+        image name: 'test.png'
+        straightened image name: 'test-straightened.png'
+    """
+    for fp in rw.flipbook_pages:
+        lab_frame_image = fp[0].data
+        center_tck, width_tck = fp.annotations['pose']
+        if center_tck is not None:
+            warp = worm_spline.to_worm_frame(lab_frame_image, center_tck, width_tck)
+            warp = colorize.scale(warp).astype(numpy.uint8)
```

Comment:
Why not just save out the warped worm in the original dtype? (uint8 or 16) and then save the mask separately? It's far from clear that scaling each worm to its own min/max is the right thing to do. You could have separate CLI programs to scale worms, combine masks into an alpha channel, and so forth.

If you really want to write out scaled images only, I would suggest using code like that in `elegant/gui/compose_images.py` to gather up all the flipbook layers, warp each one, and then scale and composite them as per the layer preferences. I could help here if it's not clear what exactly to do -- just add it as a `TODO` in a comment...